components/post_view/message_attachments/action_menu/action_menu.tsx

```diff
@@ -1,29 +1,78 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 
+import {DispatchFunc, GetStateFunc} from 'mattermost-redux/types/actions';
+import {UserProfile} from 'mattermost-redux/types/users';
+import {Channel} from 'mattermost-redux/types/channels';
+import {ServerError} from 'mattermost-redux/types/errors';
+
 import MenuActionProvider from 'components/suggestion/menu_action_provider';
 import GenericUserProvider from 'components/suggestion/generic_user_provider.jsx';
 import GenericChannelProvider from 'components/suggestion/generic_channel_provider.jsx';
 import AutocompleteSelector from 'components/autocomplete_selector';
 import PostContext from 'components/post_view/post_context';
 
-export default class ActionMenu extends React.PureComponent {
-    static propTypes = {
-        postId: PropTypes.string.isRequired,
-        action: PropTypes.object.isRequired,
-        selected: PropTypes.object,
-        disabled: PropTypes.bool,
-        actions: PropTypes.shape({
-            autocompleteChannels: PropTypes.func.isRequired,
-            selectAttachmentMenuAction: PropTypes.func.isRequired,
-            autocompleteUsers: PropTypes.func.isRequired,
-        }).isRequired,
-    }
-
-    constructor(props) {
+type Error = ServerError & {id: string};
```

Comment:
Is this something we can rely on? I am not experienced at all with this specific type, but I wonder what would happen if there are any changes in `ServerError`. I guess it would break this `Error` definition (not sure why we need the `id` in the first place).

Reply:
Thank you for the clarification! I guess this is safe, although still a bit weird (not because of the PR, but because of how errors are handled elsewhere).